resolwe_bio/processes/import_data/ml_ready_expressions.py

```diff
@@ -0,0 +1,107 @@
+"""Upload ML-ready expression matrix."""
+import os
+
+import pandas as pd
+
+from resolwe.process import DataField, FileField, Process, SchedulingClass, StringField
+from resolwe.process.models import Entity
+
+
+class UploadMLExpression(Process):
+    """Upload ML-ready expression matrix."""
+
+    slug = "upload-ml-expression"
+    name = "ML-redy expression"
+    process_type = "data:ml:table:expression"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/common:3.0.1"}
+        },
+        "resources": {
+            "cores": 1,
+            "memory": 8192,
+            "storage": 10,
+        },
+    }
+    data_name = "{{ reference_space|name }}"
+
+    class Input:
+        """Inputs."""
+
+        exp = FileField(
+            label="Transformed expressions",
+            description="A TAB separated file containing transformed "
+            "expression values with sample IDs for index (first column "
+            "with label sample_id) and ENSEMBL IDs (recommended but not "
+            "required) for the column names.",
+        )
+        source = StringField(label="Feature source")
+        species = StringField(label="Species")
+        reference_space = DataField("ml:space", label="Reference ML space")
+
+    class Output:
+        """Outputs."""
+
+        exp = FileField(label="Transformed expressions")
```

Comment:
Should a `exp_json` output be added for a possible utility in FE apps?

Reply:
AFAIK, this process is just for storing exp matrices. So I would rather avoid this.

Reply:
After consulting with @mstajdohar the decision remains to not add json fields. If this data will be necessary in visualisation we will make a dedicated tabale in db to support fast retrieval of expressions values.